readthedocs/search/management/commands/reindex_elasticsearch.py

```diff
@@ -7,69 +7,73 @@
 from django.core.management import BaseCommand
 from django_elasticsearch_dsl.registries import registry
 
-from ...tasks import (index_objects_to_es_task, switch_es_index_task, create_new_es_index_task,
-                      index_missing_objects_task)
-from ...utils import chunks
+from ...tasks import (index_objects_to_es, switch_es_index, create_new_es_index,
+                      index_missing_objects)
+from ...utils import chunk_queryset
 
 log = logging.getLogger(__name__)
 
 
 class Command(BaseCommand):
 
     @staticmethod
-    def _get_indexing_tasks(app_label, model_name, instance_ids, document_class, index_name):
-        chunk_objects = chunks(instance_ids, settings.ES_TASK_CHUNK_SIZE)
+    def _get_indexing_tasks(app_label, model_name, queryset, document_class, index_name):
+        queryset = queryset.values_list('id', flat=True)
+        chunked_queryset = chunk_queryset(queryset, settings.ES_TASK_CHUNK_SIZE)
 
-        for chunk in chunk_objects:
+        for chunk in chunked_queryset:
             data = {
                 'app_label': app_label,
                 'model_name': model_name,
                 'document_class': document_class,
                 'index_name': index_name,
-                'objects_id': chunk
+                'objects_id': list(chunk)
             }
-            yield index_objects_to_es_task.si(**data)
+            yield index_objects_to_es.si(**data)
 
     def _run_reindex_tasks(self, models):
         for doc in registry.get_documents(models):
-            qs = doc().get_queryset()
-            instance_ids = list(qs.values_list('id', flat=True))
+            queryset = doc().get_queryset()
+            # Get latest object from the queryset
+            latest_object = queryset.latest('modified_date')
+            latest_object_time = latest_object.modified_date
```

Comment:
This should just use the current time, not query the entire queryset for the time. This will be quite a slow query likely in prod.

Reply:
Yeah! I was also thinking about it.